src/map/script.c

```diff
@@ -114,9 +114,36 @@ static bool script_nick2sd_(struct script_state *st, uint8 loc, struct map_sessi
 	return (*sd) ? true : false;
 }
 
+/**
+ * Get `bl` from an ID in `loc` param, if ID is 0 will return the `bl` of the script's activator.
+ * @param st Script
+ * @param loc Location to look for ID in script parameter
+ * @param bl Variable that will be assigned
+ * @return True if `bl` is assigned, false otherwise
+ **/
+static bool script_rid2bl_(struct script_state *st, uint8 loc, struct block_list **bl, const char *func) {
+	if (script_hasdata(st, loc)) {
+		int unit_id = script_getnum(st, loc);
+		if (!unit_id)
+		{
+			if (!(*bl = map_id2bl(st->rid)))
+				ShowError("%s: Unit with ID '%d' is not found.\n", func, unit_id);
+		}
+		else
+		{
+		if (!(*bl = map_id2bl(unit_id)))
+				ShowError("%s: Unit with ID '%d' is not found.\n", func, unit_id);
+		}
+	}
+	else
+		*bl = NULL;
```

Comment:
else case should possibly also use the attached unit.

in reference to this and since you want to support 0 as "use attached bl" i'd do it somewhat like this:

```
int unit_id;

if( !script_hasdata(st, loc) || ( unit_id = script_getnum(st_loc) ) == 0 ){
	unit_id = st->rid;
}

*bl = map_id2bl(unit_id);
```